Tests/AccountTests/Push/AutopushTests.swift

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import XCTest
+import MozillaAppServices
+import Account
+
+class AutopushTests: XCTestCase {
+    private var mockPushManager = MockPushManager()
+    private var autopushClient: Autopush?
+
+    override func setUp() async throws {
+        autopushClient = Autopush(withPushManager: mockPushManager)
+        try await super.setUp()
```

Comment:
Curious, was that `try await super.setUp()` needed? 

We normally just do:
```
    override func setUp() {
        super.setUp()
        // set variables for tests
    }
```

We also use the `setUp` method to create instances. This means in your case`private var mockPushManager = MockPushManager()` should be created in `setUp` instead. To do so, we just use a force `!` on the variable. 

```
    private var mockPushManager: MockPushManager!
    private var autopushClient: Autopush! // Same for autopushClient var

    override func setUp() {
        super.setUp()
        mockPushManager = MockPushManager()
        autopushClient = Autopush(withPushManager: mockPushManager)
    }
```

We normally avoid `!` in production code, but this is fine and normal for unit tests.

Reply:
Aaah! I like the feedback for the setup and teardown - 

the `try await super.setUp()` was needed because the `setUp` function I'm using is an `override func setUp() async throws` so calling `super.setUp` ends up calling an async throwing function

I like using `!` in this case, and nil'ing out the `mockPushManager` in the teardown so I'll do that 👀

Reply:
I'll also add that i'm using the `async` setup function because the constructor for `Autopush` is async

Reply:
eeeh I take that back, I'm using an alternative constructor here, meaning there is no need to use the async setup 🤦 An artifact of previously using the async constructor